src/trio/_tests/test_sync.py

```diff
@@ -586,3 +589,59 @@ async def lock_taker() -> None:
         await wait_all_tasks_blocked()
         assert record == ["started"]
         lock_like.release()
+
+
+async def test_lock_acquire_unowned_lock() -> None:
+    """Test that trying to acquire a lock whose owner has exited raises an error.
+    Partial fix for https://github.com/python-trio/trio/issues/3035
```

Comment:
How about testing the full fix because this PR should be the full fix.

Reply:
this+next test is the full fix, so the docstring is outdated. Thanks

Although that issue contains some other loose thoughts (give warning if exiting without releasing the lock, semaphore things that got set aside, add this feature to capacitylimiter/semaphore) that should perhaps be mulled over before closing it.